qiskit/quantum_info/states/statevector.py

```diff
@@ -113,6 +113,72 @@ def __init__(
                 raise QiskitError("Invalid input: not a vector or column-vector.")
         super().__init__(op_shape=OpShape.auto(shape=shape, dims_l=dims, num_qubits_r=0))
 
+    @classmethod
+    def from_circuit(cls, circuit, ignore_set_layout=False):
+        """Create a Statevector from a quantum circuit.
+
+        Args:
+            circuit (QuantumCircuit): A quantum circuit
+            ignore_set_layout (bool): When set to ``True``, if the input ``circuit``
+                has a layout set, it will be ignored. Defaults to ``False``.
+
+        Returns:
+            Statevector: The statevector representing the circuit evolution
+
+        Example:
+            Create a statevector from a transpiled circuit:
+
+            .. code-block:: python
```

Comment:
We could use 
```suggestion    
            .. plot::
               :include-source:
               :nofigs:
```
to run and test the snippet